### PR TITLE
[PM-6654] new app id service for angular

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Migrates appId and anonymous appId to state providers

## Code changes

- **services**: removes bgService call
- **utils**: I wanted to use our guid regex is a test, so extracting it here was nice
- **app id service **: we no longer directly access storage here, and just define this as a global item
- **global state / user state**: I realized our returns documentation doesn't really imply that my use case in appIdService is valid, but `update` is guaranteed to return whatever `state` is at the resolution of the `update` call, even if the backing store isn't updated, making it perfect for this annoying get and set use case.
- **migrations**: Migrated the directly stored values to global state provider

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: libs/common/src/platform/services/app-id.service.ts [hunk 1]
```diff
@@ -1,31 +1,28 @@
...
+  private appIdState = this.globalStateProvider.get(APP_ID_KEY);
+  private anonymousAppIdState = this.globalStateProvider.get(ANONYMOUS_APP_ID_KEY);
+  constructor(private globalStateProvider: GlobalStateProvider) {}
 
-  private async makeAndGetAppId(key: string) {
-    const existingId = await this.storageService.get<string>(key, {
-      htmlStorageLocation: HtmlStorageLocation.Local,
+  async getAppId(): Promise<string> {
+    return await this.appIdState.update(() => Utils.newGuid(), {
+      shouldUpdate: (value) => value == null,
```
Comment (justindbaur) [inline libs/common/src/platform/services/app-id.service.ts:19]: This is quite possibly the most perfect use of `shouldUpdate` and the return value of `update` which is why the following comment kinda hurts me to say: Should this be returning in observable instead? Maybe (not tested):

```typescript
appId$ = this.appIdState.state$.pipe(
  tap(async (appId) => {
    if (appId == null) {
      await this.appIdState.update(() => Utils.newGuid());
    }
  }),
  filter(appId != null),
);
```

Definitely kinda wacky but is more reactive.

**Review by justindbaur — COMMENTED** [review]
Just one thought about making it more reactive, it does change the api surface and I'm not entirely sure how this is consumed so there is a chance this could never be used reactively so if you disagree I am happy to approve.

File: libs/common/src/platform/services/app-id.service.ts (hunk 1 above)
Comment (MGibson1) [reply libs/common/src/platform/services/app-id.service.ts:19]: I like that, pretty clever use of filter to remove the updated versions and rely on the update to trigger a second emit.

I don't know how useful a reactive interface will be here, but there's certainly no harm in adding one.

**Review by justindbaur — APPROVED** [review]

**Review by justindbaur — APPROVED** [review]

File: libs/common/src/platform/services/app-id.service.ts (hunk 1 above)
Comment (justindbaur) [reply libs/common/src/platform/services/app-id.service.ts]: Yeah a quick glance shows this isn't shown in any UI at all but it's the principle I guess.

**Pull request merged**